src/items/time.rs

```diff
@@ -0,0 +1,843 @@
+//! > ahead of UTC (e.g., India). This is the best way to specify a time zone
+//! > correction by fractional parts of an hour. The maximum zone correction is
+//! > 24 hours.
+//! >
+//! > Either ‘am’/‘pm’ or a time zone correction may be specified, but not both.
+
+use std::fmt::Display;
+
+use chrono::FixedOffset;
+use winnow::{
+    ascii::{dec_uint, digit1, float},
+    combinator::{alt, opt, peek, preceded},
+    error::{AddContext, ContextError, ErrMode, StrContext},
+    seq,
+    stream::AsChar,
+    token::take_while,
+    PResult, Parser,
+};
+
+use super::{relative, s};
+
+#[derive(PartialEq, Debug, Clone, Default)]
+pub struct Offset {
+    pub(crate) negative: bool,
+    pub(crate) hours: u32,
+    pub(crate) minutes: u32,
+}
+
+#[derive(PartialEq, Clone, Debug, Default)]
+pub struct Time {
+    pub hour: u32,
+    pub minute: u32,
+    pub second: f64,
+    pub offset: Option<Offset>,
+}
+
+impl Offset {
+    fn merge(self, offset: Offset) -> Option<Offset> {
+        fn combine(a: u32, neg_a: bool, b: u32, neg_b: bool) -> (u32, bool) {
+            if neg_a == neg_b {
+                (a + b, neg_a)
+            } else if a > b {
+                (a - b, neg_a)
+            } else {
+                (b - a, neg_b)
+            }
+        }
+        let (hours_minutes, negative) = combine(
+            self.hours * 60 + self.minutes,
+            self.negative,
+            offset.hours * 60 + offset.minutes,
+            offset.negative,
+        );
+        let hours = hours_minutes / 60;
+        let minutes = hours_minutes % 60;
+
+        Some(Offset {
+            negative,
+            hours,
+            minutes,
+        })
+    }
+}
+
+impl From<Offset> for chrono::FixedOffset {
+    fn from(
+        Offset {
+            negative,
+            hours,
+            minutes,
+        }: Offset,
+    ) -> Self {
+        let secs = hours * 3600 + minutes * 60;
+
+        if negative {
+            FixedOffset::west_opt(secs.try_into().expect("secs overflow"))
+                .expect("timezone overflow")
+        } else {
+            FixedOffset::east_opt(secs.try_into().unwrap()).unwrap()
+        }
+    }
+}
+
+impl Display for Offset {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(
+            fmt,
+            "{}{:02}:{:02}",
+            if self.negative { "-" } else { "+" },
+            self.hours,
+            self.minutes
+        )
+    }
+}
+
+#[derive(Clone)]
+enum Suffix {
+    Am,
+    Pm,
+}
+
+pub fn parse(input: &mut &str) -> PResult<Time> {
+    alt((am_pm_time, iso)).parse_next(input)
+}
+
+/// Parse an ISO 8601 time string
+///
+/// Also used by the [`combined`](super::combined) module
+pub fn iso(input: &mut &str) -> PResult<Time> {
+    alt((
+        (hour24, timezone).map(|(hour, offset)| Time {
+            hour,
+            minute: 0,
+            second: 0.0,
+            offset: Some(offset),
+        }),
+        seq!( Time {
```

Comment:
```suggestion
        seq!(Time {
```
The tiniest of nitpicks 😄